src/test/rbd_mirror/image_replayer/test_mock_BootstrapRequest.cc

```diff
@@ -1190,6 +1190,35 @@ TEST_F(TestMockImageReplayerBootstrapRequest, ReplayRequiresRemoteImage) {
   ASSERT_EQ(0, ctx.wait());
 }
 
+TEST_F(TestMockImageReplayerBootstrapRequest, CancelWhenRemotePrimaryDeleted) {
+  InSequence seq;
+
+  // prepare local image
+  MockPrepareLocalImageRequest mock_prepare_local_image_request;
+  MockStateBuilder mock_state_builder;
+  expect_send(mock_prepare_local_image_request, mock_state_builder,
+              m_local_image_ctx->id, m_local_image_ctx->name, 0);
+
+  // prepare remote image
+  MockPrepareRemoteImageRequest mock_prepare_remote_image_request;
+  expect_send(mock_prepare_remote_image_request, mock_state_builder,
+              "remote mirror uuid", m_remote_image_ctx->id, -ENOENT);
+  expect_is_local_primary(mock_state_builder, false);
+//  expect_is_remote_primary(mock_state_builder, true);
+  expect_is_linked(mock_state_builder, true);
+  expect_is_linked(mock_state_builder, true);
```

Comment:
I don't think the second `expect_is_linked()` call is needed.  You are probably running with verbose logs enabled so that's why it popped up for you.